chore(docs): add asynchronous example to remark-plugin-tutorial (#26116)
gatsbyjs/gatsby: docs/tutorial/remark-plugin-tutorial.md

A small library [mdast-util-to-string](https://github.com/syntax-tree/mdast-util-to-string) by Unified was used to extract the plain text of the inner nodes. This would remove links or other types of nodes inside the heading, but given you have full access to the markdown AST, you can modify it however you wish.

## Using asynchronous behavior

> Gatsby supports the usage of asynchronous behavior in plugins, and gatsby-transformer-remark`uses [unified](https://github.com/unifiedjs/unified) under the hood, so in this case the`gatsby-remark-purple-headers`transformer can be converted to asynchronous by adding the`async` keyword to the function declaration.

Reviewer: ```suggestion
Gatsby supports the usage of asynchronous behavior in plugins, and `gatsby-transformer-remark` uses [unified](https://github.com/unifiedjs/unified) under the hood. The following example shows how the `gatsby-remark-purple-headers` transformer can be converted to asynchronous by adding the`async` keyword to the function declaration.
```

Author: @laurieontech I committed these changes along with the addition of the markdown code title in ec410d6. Thanks!

> ```js

Reviewer: Which file would this be? `gatsby-node.js`?

Author: The file would be `plugins/gatsby-remark-purple-headers/index.js`. I'm adding this to the title tag of the associated markdown section.

> Gatsby supports the usage of asynchronous behavior in plugins, and gatsby-transformer-remark`uses [unified](https://github.com/unifiedjs/unified) under the hood. The following example shows how the`gatsby-remark-purple-headers`transformer can be converted to asynchronous by adding the`async` keyword to the function declaration.

Reviewer 2: ```suggestion
Gatsby supports the usage of asynchronous behavior in plugins, and `gatsby-transformer-remark` uses [unified](https://github.com/unifiedjs/unified) under the hood. The following example shows how the `gatsby-remark-purple-headers` transformer can be converted to asynchronous by adding the `async`  keyword to the function declaration.
```

const visit = require("unist-util-visit")
const toString = require("mdast-util-to-string")

// highlight-next-line
module.exports = async ({ markdownAST }, pluginOptions) => {
  visit(markdownAST, "heading", node => {
    let { depth } = node

    // Skip if not an h1
    if (depth !== 1) return

    // Grab the innerText of the heading node
    let text = toString(node)

    const html = `
        <h1 style="color: rebeccapurple">
          ${text}
        </h1>
      `

    node.type = "html"
    node.children = undefined
    node.value = html
  })

  return markdownAST
}
```

> 

Reviewer 2: ```suggestion
A real-world example of this would be [`gatsby-remark-code-repls`](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-remark-code-repls/src/gatsby-node.js).

```
